### PR TITLE
Simplifying genmodel's code a little; fixing a bug in genmodel's calc…

### PR DESCRIPTION
**Pull request by st-pasha**
…ulation of n_effective_trees that was affecting old drf mojos

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/517)

### CODE REVIEW

File: h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java [hunk 1]
```diff
@@ -53,32 +52,4 @@ public GbmMojoModel(String[] columns, String[][] domains) {
         return score0(row, 0.0, preds);
     }
 
-    /**
```
Comment (tomkraljevic) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java]: It took me a while to figure out what was going on in terms of the number of compressed trees per class.  Which is why I added this comment.

It looks like you are trying to unify the code in the shared class, which is good, but you should add a public accessor function with a good comment and make the actual field private.

File: h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java [hunk 2]
```diff
@@ -15,15 +13,8 @@ protected void readModelData() throws IOException {
...
-      if (_model instanceof DrfMojoModel) {
-        tpc = ((DrfMojoModel) _model)._effective_n_classes;
-      }
-      else if (_model instanceof GbmMojoModel) {
-        tpc = ((GbmMojoModel) _model).calcNClassesToScore();
-      }
-      else {
-        throw new RuntimeException("Unknown Mojo Model");
-      }
+      Boolean bdt = readkv("binomial_double_trees");  // This flag exists only for DRF models
```
Comment (st-pasha) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java]: As you may notice, the old code contained a bug: the `tpc = ((DrfMojoModel) _model)._effective_n_classes;` was executing at the moment when `DrfMojoReader` hadn't read the model information yet, and therefore `_effective_n_classes` will be 0.

We didn't keep any old DRF models around (before the `n_trees_per_class` variable was added), so this error was not caught ...

**Comment by st-pasha** [comment]
@tomkraljevic Added documentation as you requested. Also a new function `treeIndex(i, j)` for a more reliable calculation of tree index.

**Comment by st-pasha** [comment]
@tomkraljevic Perhaps you're right and we should wait a bit before merging, so that we can test this change against the new testing framework.

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Pull request merged**